### PR TITLE
Calculator

### PR DESCRIPTION
**Pull request by marioalej**
Improved calculator

### CODE REVIEW

File: app/src/test/java/com/globant/mariorangel/kotlinonboarding/mvp/presenter/CountPresenterTest.kt [hunk 1]
```diff
@@ -1,62 +1,29 @@
...
     @Test
-    fun isShouldIncCountByOne() {
-        model!!.reset()
-        RxBus.post(OnCountButtonPressedBusObserver.OnCountButtonPressed())
-
-        val count = "1"
-        assertEquals(model!!.count, 1)
-        verify(view).setCount(count)
+    fun shouldPlusFirstAndSecondNumber() {
+        //Mockito.`when`(model.addition("10", ))
```
Comment (gaston-mira-glb) [inline app/src/test/java/com/globant/mariorangel/kotlinonboarding/mvp/presenter/CountPresenterTest.kt]: don't keep commented code please.

**Review by gaston-mira-glb — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/globant/mariorangel/kotlinonboarding/mvp/model/CountModel.kt [hunk 2]
```diff
@@ -1,16 +1,43 @@
 package com.globant.mariorangel.kotlinonboarding.mvp.model
 
+import com.globant.mariorangel.kotlinonboarding.mvp.view.CountView
```
Comment (nacho-globant) [inline app/src/main/java/com/globant/mariorangel/kotlinonboarding/mvp/model/CountModel.kt]: remove this pls

File: app/src/main/java/com/globant/mariorangel/kotlinonboarding/mvp/view/CountView.kt [hunk 3]
```diff
@@ -1,24 +1,77 @@
...
-        activity!!.count_label.text = count
+    /**
+     * will update the view with the operation result
+     */
+    fun setResult(result: String) {
+        activity!!.count_result.text = result
+        activity!!.clear_button.isEnabled = true
     }
+
+    companion object {
```
Comment (nacho-globant) [inline app/src/main/java/com/globant/mariorangel/kotlinonboarding/mvp/view/CountView.kt]: move this into model

File: app/src/main/java/com/globant/mariorangel/kotlinonboarding/mvp/view/CountView.kt [hunk 4]
```diff
@@ -1,24 +1,77 @@
...
 import kotlinx.android.synthetic.main.activity_main.*
 
 class CountView(activity: Activity) : ActivityView(activity) {
 
     init {
-        activity.count_button.setOnClickListener {
-            RxBus.post(OnCountButtonPressedBusObserver.OnCountButtonPressed())
+
+        activity.plus_button.setOnClickListener {
+            if (!activity.count_input.text.isEmpty())
```
Comment (nacho-globant) [inline app/src/main/java/com/globant/mariorangel/kotlinonboarding/mvp/view/CountView.kt]: move this condition to presenter

File: app/src/main/java/com/globant/mariorangel/kotlinonboarding/mvp/view/CountView.kt [hunk 5]
```diff
@@ -1,24 +1,77 @@
...
+                RxBus.post(OperatorButtonPressedObserver.OnButtonPressed(
+                        activity.count_input.text.toString(),
+                        PLUS
+                ))
+        }
+
+        activity.minus_button.setOnClickListener {
+            if (!activity.count_input.text.isEmpty())
```
Comment (nacho-globant) [inline app/src/main/java/com/globant/mariorangel/kotlinonboarding/mvp/view/CountView.kt]: the same

File: app/src/main/java/com/globant/mariorangel/kotlinonboarding/mvp/view/CountView.kt [hunk 6]
```diff
@@ -1,24 +1,77 @@
...
+                RxBus.post(OperatorButtonPressedObserver.OnButtonPressed(
+                        activity.count_input.text.toString(),
+                        MINUS
+                ))
+        }
+
+        activity.multiplier_button.setOnClickListener {
+            if (!activity.count_input.text.isEmpty())
```
Comment (nacho-globant) [inline app/src/main/java/com/globant/mariorangel/kotlinonboarding/mvp/view/CountView.kt]: the same

File: app/src/main/java/com/globant/mariorangel/kotlinonboarding/mvp/view/CountView.kt [hunk 7]
```diff
@@ -1,24 +1,77 @@
...
+                RxBus.post(OperatorButtonPressedObserver.OnButtonPressed(
+                        activity.count_input.text.toString(),
+                        MULTIPLIER
+                ))
+        }
+
+        activity.divider_button.setOnClickListener {
+            if (!activity.count_input.text.isEmpty())
```
Comment (nacho-globant) [inline app/src/main/java/com/globant/mariorangel/kotlinonboarding/mvp/view/CountView.kt]: the same

File: app/src/main/java/com/globant/mariorangel/kotlinonboarding/mvp/view/CountView.kt [hunk 8]
```diff
@@ -1,24 +1,77 @@
...
+                RxBus.post(OperatorButtonPressedObserver.OnButtonPressed(
+                        activity.count_input.text.toString(),
+                        DIVIDER
+                ))
         }
 
-        activity.reset_button.setOnClickListener {
-            RxBus.post(OnResetButtonPressedBusObserver.OnResetButtonPressed())
+        activity.equal_button.setOnClickListener {
+            if (!activity.count_input.text.isEmpty())
```
Comment (nacho-globant) [inline app/src/main/java/com/globant/mariorangel/kotlinonboarding/mvp/view/CountView.kt]: the same

File: app/src/main/java/com/globant/mariorangel/kotlinonboarding/mvp/presenter/CountPresenter.kt [hunk 9]
```diff
@@ -1,32 +1,88 @@
...
+        // Subscribe listener for the equal button
+        RxBus.subscribe(activity, object : EqualButtonPressedObserver() {
+            override fun onEvent(value: OnEqualButtonPressed) {
+                if (!activity.count_input.text.isEmpty())
+                    makeOperation(value.value.toDouble())
+            }
+        })
+    }
+
+    fun checkEmptyText() {
```
Comment (nacho-globant) [inline app/src/main/java/com/globant/mariorangel/kotlinonboarding/mvp/presenter/CountPresenter.kt]: Remove if you dont use pls

File: app/src/main/java/com/globant/mariorangel/kotlinonboarding/mvp/view/CountView.kt [hunk 10]
```diff
@@ -1,24 +1,65 @@
 package com.globant.mariorangel.kotlinonboarding.mvp.view
 
 import android.app.Activity
-import com.globant.mariorangel.kotlinonboarding.util.bus.OnCountButtonPressedBusObserver
-import com.globant.mariorangel.kotlinonboarding.util.bus.OnResetButtonPressedBusObserver
+import com.globant.mariorangel.kotlinonboarding.mvp.model.CountModel
```
Comment (nacho-globant) [inline app/src/main/java/com/globant/mariorangel/kotlinonboarding/mvp/view/CountView.kt]: The view should not know the model.

Move the constant to another class/interface

**Pull request merged**